src/series.jl

```diff
@@ -519,16 +519,31 @@ split_kw(key, val::Tuple, indices) = Tuple(split_kw(key, v, indices) for v in va
 # split the group into 1 series per group, and set the label and idxfilter for each
 @recipe function f(groupby::GroupBy, args...)
     lengthGroup = maximum(union(groupby.groupIds...))
-    for (i,glab) in enumerate(groupby.groupLabels)
-        @series begin
-            label     --> string(glab)
-            idxfilter --> groupby.groupIds[i]
-            for (key,val) in d
-                if splittable_kw(key, val, lengthGroup)
-                    :($key) := split_kw(key, val, groupby.groupIds[i])
+    if !(RecipesBase.group_as_matrix(args[1]))
+        for (i,glab) in enumerate(groupby.groupLabels)
+            @series begin
+                label     --> string(glab)
+                idxfilter --> groupby.groupIds[i]
+                for (key,val) in d
+                    if splittable_kw(key, val, lengthGroup)
+                        :($key) := split_kw(key, val, groupby.groupIds[i])
+                    end
                 end
+                args
             end
-            args
         end
+    else
+        g = args[1]
+        x, y = g.args
```

Comment:
What about z? What if there is only y?

Reply:
The "WIP" label is for real: I just got this to work and wanted to see if we're happy with the approach, I still need to:
- Figure it out in case of only `y` or `x`, `y`, `z`
- Add the same fix for "column like" keyword arguments as in here: #1033